src/parser/jobs/brd/modules/Snapshots.tsx

```diff
@@ -0,0 +1,197 @@
+
+	private targets = new Map<string, AbilityEvent[]>()
+
+	protected init() {
+		this.addEventHook('cast', {by: 'player', abilityId: SNAPSHOTTERS}, this._onSnapshot)
+		this.addEventHook('normaliseddamage', {by: 'player', abilityId: DOT_STATUSES}, this._onDotTick)
+		this.addEventHook(['applydebuff', 'refreshdebuff'], this._onApply)
+		this.addEventHook('removedebuff', this._onRemove)
+	}
+
+	private _onApply(event: AbilityEvent) {
+		const targetKey = `${event.targetID}-${event.targetInstance}`
+
+		if (!this.targets.has(targetKey)) {
+			this.targets.set(targetKey, [])
+		}
+
+		const target = this.targets.get(targetKey)
+		if (target) {
+			const index = target.findIndex(statusEvent =>
+				statusEvent.ability.guid === event.ability.guid)
+
+			if (index === -1) {
+				target.push(event)
+			}
+		}
+	}
+
+	private _onRemove(event: AbilityEvent) {
+		const targetKey = `${event.targetID}-${event.targetInstance}`
+		const targetStatuses = this.targets.get(targetKey)
+
+		if (targetStatuses) {
+			const index = targetStatuses.findIndex(status =>
+				status.ability.guid === event.ability.guid)
+
+			if (index > -1) {
+				targetStatuses.splice(index, 1)
+			}
+		}
+	}
+
+	private _getStatuses(event: CastEvent) {
+		const targetKey = `${event.targetID}-${event.targetInstance}`
+		const playerStatuses = this.combatants.selected.getStatuses().map(
+			(status: AbilityEvent) => status.ability.guid)
+		const targetStatuses = this.targets.get(targetKey)?.map(
+			(status: AbilityEvent) => status.ability.guid) || []
+
+		return [...playerStatuses, ...targetStatuses]
+	}
+
+	private _onSnapshot(event: CastEvent) {
+		if (this.currentSnapshot) {
+			this.snapshots.push(this.currentSnapshot)
+		}
+
+		this.currentSnapshot = {
+			snapEvent: event,
+			statusIDs: this._getStatuses(event),
+			ticks: [],
+		}
+	}
+
+	private _onDotTick(event: NormalisedDamageEvent) {
+		// This will come in handy in the future
+		this.currentSnapshot?.ticks.push(event)
+	}
+
+	output() {
+		if (this.snapshots.length === 0) {
+			return
+		}
+
+		// Builds a row for each snapshot event
+		const rows = this.snapshots.map(snap => {
+
+			snap.statusIDs.sort()
+
+			// Move personal buffs to the front of the status list
+			snap.statusIDs.map((status, index) => {
+				if (PERSONAL_STATUSES.includes(status)) {
+					snap.statusIDs.unshift(
+						snap.statusIDs.splice(index, 1)[0],
+					)
+				}
+			})
+
+			const snapshotDotCell = <Table.Cell>
+				{
+					snap.statusIDs.map(id => {
+						const status = getDataBy(STATUSES, 'id', id)
+						if (status && DOT_STATUSES.includes(id)) {
+							return <StatusLink key={status.name} showName={false} iconSize="35px" {...status}/>
+						}
+					})
+				}
+			</Table.Cell>
+
+			const snapshotBuffCell = <Table.Cell>
+				{
+					snap.statusIDs.map(id => {
+						// Avoid showing statuses we do not currently know of and statuses known not to affect bard DoTs
+						const status = getDataBy(STATUSES, 'id', id)
```

Comment:
Same deal here, plus could just filter when adding stuff to the snapshot in the first place.